### PR TITLE
Feat/cascader search

### LINKED ISSUE #2098: [Cascader] Ref method support search
### Which Component Need Enhancement 期望新增功能的组件
- Casacder

### What does the proposed API look like 期望支持的API
- Ref method 暴露出搜索相关的function，允许用户主动调用

### Feature Description 功能描述
用于实现类似 UI 时，配合 topSlot 使用
image

### PR DESCRIPTION
**Pull request by YyumeiZhang**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [x] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Fixes #2098 

### Changelog
🇨🇳 Chinese
- Feat: Cascader 支持通过 ref 调用 search 方法 #2098 

---

🇺🇸 English
- Feat: Cascader supports calling the search method through ref #2098 

### Checklist
- [x] Test or no need
- [x] Document or no need
- [x] Changelog or no need

### Other
- [ ] Skip Changelog

### CODE REVIEW

File: content/input/cascader/index-en-US.md [hunk 1]
```diff
@@ -1855,6 +1855,7 @@ function Demo() {
 | preventScroll | Indicates whether the browser should scroll the document to display the newly focused element, acting on the focus method inside the component, excluding the component passed in by the user                                                 | boolean | - | 2.15.0 |
 |restTagsPopoverProps | The configuration properties of the [Popover](/en-US/show/popover#API%20Reference)                                                                                                                                                            |PopoverProps   | {}  |1.28.0|
 | searchPlaceholder  | Placeholder for search input                                                                                                                                                                                                                  | string | - | -  |
+| searchRender         | Whether to display the search input box | boolean | true | 2.54.0 |
```
Comment (pointhalo) [inline content/input/cascader/index-en-US.md:1858]: 保持 showXXX 的命名风格？

Comment (YyumeiZhang) [reply content/input/cascader/index-en-US.md:1858]: 有点犹豫的原因在于 tree中用的是 searchRender（同时支持 function 和 boolean）。
image
我们应该不会用到 函数模式，我修改为 showSearchInTrigger？

Comment (pointhalo) [reply content/input/cascader/index-en-US.md:1858]: 那感觉跟 TreeSelect对齐才更合理。 searchPosition：custom、trigger
默认是trigger，如果custom就相当于用户自行控制input渲染。未来可以有需要再加 'dropdown'

Comment (pointhalo) [reply content/input/cascader/index-en-US.md:1858]: image

**Pull request merged**